core/src/main/scala/cats/data/IndexedStateT.scala

```diff
@@ -167,6 +167,13 @@ final class IndexedStateT[F[_], SA, SB, A](val runF: F[SA => F[(SB, A)]]) extend
    */
   def get(implicit F: Functor[F]): IndexedStateT[F, SA, SB, SB] =
     inspect(identity)
+
+  def flatTap[B, SC](fas: A => IndexedStateT[F, SB, SC, B])(implicit F: FlatMap[F]): IndexedStateT[F, SA, SC, A] =
```

Comment:
Here too.

Reply:
This version allows changing the state type. `Monad` instance exists for SA = SB only.